pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -203,10 +234,58 @@ func (o *MustGatherOptions) completeImages() error {
 		}
 		o.Images = append(o.Images, image)
 	}
+	if o.AllImages {
+		// find all csvs and clusteroperators with the annotation "operators.openshift.io/must-gather-image"
+		pluginImages := make(map[string]struct{})
+		var err error
+
+		pluginImages, err = o.annotatedCSVs()
+		if err != nil {
+			return err
+		}
+
+		cos, err := o.ConfigClient.ConfigV1().ClusterOperators().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, item := range cos.Items {
+			ann := item.GetAnnotations()
+			if v, ok := ann[mgAnnotation]; ok {
+				pluginImages[v] = struct{}{}
+			}
+		}
+		// delete the default image to avoid duplication in case an Operator had it in its annotation
+		delete(pluginImages, o.Images[0])
```

Comment:
What about the case where the default image is not added through annotations?

Reply:
Sorry, I don't follow you. By "default image", are we both referring to below:
```go
image = "registry.redhat.io/openshift4/ose-must-gather:latest"
```
If yes, the only place I see it being added to `o.Images` is in the block above:
```go
if len(o.Images) == 0 || o.AllImages
```
If none of the annotations are adding the same image, the `delete` operation would delete nothing. Are you suggesting adding a check like:
```go
if _, ok := pluginImages[o.Images]; ok {
    delete(pluginImages, o.Images[0])
}
```

Reply:
Oh, I see. That works as well.